### ISSUE TITLE
Search guard : Kibana saved object get deleted  by Read only user 

### ISSUE THREAD

**Issue by apmalla** [floragunncom/search-guard-kibana-plugin#273]
First, let me thank you  entire SEARCH GUARD  team  for that amazing contribution towards opensource community ,  Nice work! 

Configuration: 
      elastic search: 6.5.4 
          search guard: 6-6.5.4-25.1 
      kibana: 6.5.4
          search guard: 6.5.4-18.4  

# User
amulya:
  hash: $1234567890
  roles:
  - amulya

# Custom roles  
sg_amulya:
  readonly: true
  cluster:
    - CLUSTER_COMPOSITE_OPS_RO
  indices:
    '*':
      '*':
        - READ

# Role mapping:
sg_amulya:
  readonly: true
  backendroles:
    - amulya
  users:
    - amulya

with the above mentioned configuration  , that user is deleting kibana saved object , like search dash board and index also from GUI 

Kindly guide me how i will give a user to read only access , so that he can only view the resources

**Comment by floragunncom** [comment]
@jochenkressin @mgustafsson1 can you have a look?

**Comment by riteshsoni10** [comment]
Facing the same issue.

**Comment by apmalla** [comment]
Multitenancy feature is the solution by storing data into different indexes . 

Questions : 
- As it is an enterprise feature , is there is any other option for achieving same goal ? 
- lets suppose an enterprise have per user access having strength 1000+ , what the solution ? 
 
@floragunncom  team could you respond ?

**Comment by jochenkressin** [comment]
Yes, there is an alternative to MT if you just need users with RW and other users with RO access.

In  SG7 this is relatively easy since we have made the GLOBAL tenant (the default tenant, also used of MT is not enabled) a first-class citizen.

For SG6 this requires a bit of manual configuration. Have a look the role definition for the sg_kibana_user role:

```
sg_kibana_user:
  ...
  cluster:
    - INDICES_MONITOR
    - CLUSTER_COMPOSITE_OPS
  indices:
    '?kibana':
      '*':
        - MANAGE
        - INDEX
        - READ
        - DELETE
    '?kibana-6':
       ....
```

This is the default one, and apart from READ permission it also grants further permissions like DELETE and INDEX.

While I have not tried it hands-on myself, the first thing I would to is to create a role that has only  READ access to the indices listed in the sg_kibana_user role, like:

```
sg_kibana_user_readonly:
  ...
  cluster:
    - INDICES_MONITOR
    - CLUSTER_COMPOSITE_OPS
  indices:
    '?kibana':
      '*':
        - READ
    '?kibana-6':
      '*':
        - READ
       ....
```

Then, for the read-only Kibana users, instead of mapping them sg_kibana_user, you would map them to  sg_kibana_user_readonly.

**Comment by jochenkressin** [comment]
@riteshsoni10 In the SG7 role definitions you will find a new section called "tenant_permission". This section, amongst other things, controls access to Kibana.

Let's assume you have a role for a user that has READ access to your data:

```
sg_read_all:
  cluster_permissions:
    - "*"
  index_permissions:
    - index_patterns:
        - "*"
      allowed_actions:
        - SGS_READ
```

To limit Kibana permissions to READ as well, assign the SGS_KIBANA_ALL_READ action group to the SGS_GLOBAL_TENANT. As outlined above, this is the default tenant  (aka .kibana index):

sg_read_all:
  cluster_permissions:
    - "*"
  index_permissions:
    - index_patterns:
        - "*"
      allowed_actions:
        - SGS_READ
  tenant_permissions:
    - tenant_patterns:
        - "SGS_GLOBAL_TENANT"
      allowed_actions:
        - "SGS_KIBANA_ALL_READ"
```

**Comment by apmalla** [comment]
Thanks you so much @jochenkressin , will do a short POC and keep u posted the results .

**Comment by riteshsoni10** [comment]
> Yes, there is an alternative to MT if you just need users with RW and other users with RO access.
> 
> In SG7 this is relatively easy since we have made the GLOBAL tenant (the default tenant, also used of MT is not enabled) a first-class citizen.
> 
> For SG6 this requires a bit of manual configuration. Have a look the role definition for the sg_kibana_user role:
> 
> ```
> sg_kibana_user:
>   ...
>   cluster:
>     - INDICES_MONITOR
>     - CLUSTER_COMPOSITE_OPS
>   indices:
>     '?kibana':
>       '*':
>         - MANAGE
>         - INDEX
>         - READ
>         - DELETE
>     '?kibana-6':
>        ....
> ```
> 
> This is the default one, and apart from READ permission it also grants further permissions like DELETE and INDEX.
> 
> While I have not tried it hands-on myself, the first thing I would to is to create a role that has only READ access to the indices listed in the sg_kibana_user role, like:
> 
> ```
> sg_kibana_user_readonly:
>   ...
>   cluster:
>     - INDICES_MONITOR
>     - CLUSTER_COMPOSITE_OPS
>   indices:
>     '?kibana':
>       '*':
>         - READ
>     '?kibana-6':
>       '*':
>         - READ
>        ....
> ```
> 
> Then, for the read-only Kibana users, instead of mapping them sg_kibana_user, you would map them to sg_kibana_user_readonly.

@jochenkressin I am still unable to create a use with only read permission on kibana GUI.
I receive everything success on running sgadmin.

```
Search Guard Admin v7
Will connect to localhost:9300 ... done
Connected as CN=kirk,OU=client,O=client,L=test,C=de
Elasticsearch Version: 7.0.0
Search Guard Version: 7.0.0-35.0.0-beta1
Contacting elasticsearch cluster 'elasticsearch' and wait for YELLOW clusterstate ...
Clustername: Test_ServerCluster
Clusterstate: YELLOW
Number of nodes: 1
Number of data nodes: 1
searchguard index already exists, so we do not need to create one.
Legacy index 'searchguard' (ES 6) detected (or forced). You should migrate the configuration!
See https://docs.search-guard.com/latest/upgrading-6-7 for more details.
Populate config from /usr/share/elasticsearch/plugins/search-guard-7/sgconfig/
Will update 'sg/config' with /usr/share/elasticsearch/plugins/search-guard-7/sgconfig/sg_config.yml (legacy mode)
   SUCC: Configuration for 'config' created or updated
Will update 'sg/roles' with /usr/share/elasticsearch/plugins/search-guard-7/sgconfig/sg_roles.yml (legacy mode)
   SUCC: Configuration for 'roles' created or updated
Will update 'sg/rolesmapping' with /usr/share/elasticsearch/plugins/search-guard-7/sgconfig/sg_roles_mapping.yml (legacy mode)
   SUCC: Configuration for 'rolesmapping' created or updated
Will update 'sg/internalusers' with /usr/share/elasticsearch/plugins/search-guard-7/sgconfig/sg_internal_users.yml (legacy mode)
   SUCC: Configuration for 'internalusers' created or updated
Will update 'sg/actiongroups' with /usr/share/elasticsearch/plugins/search-guard-7/sgconfig/sg_action_groups.yml (legacy mode)
   SUCC: Configuration for 'actiongroups' created or updated
Done with success

```
After this , I had restarted kibana, but still the user test is having all permissions on kibana GUI.
